huaweicloud/services/apig/resource_huaweicloud_apig_throttling_policy.go

```diff
@@ -0,0 +1,458 @@
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile("^([\u4e00-\u9fa5A-Za-z][\u4e00-\u9fa5A-Za-z_0-9]{2,63})$"),
+					"The name contains of 3 to 64 characters, starting with a letter. Only letters, digits "+
+						"and underscore (_) are allowed."),
+			},
+			"period": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"max_api_requests": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringMatch(regexp.MustCompile("^[^<>]{1,255}$"),
+					"The description contain a maximum of 255 characters, "+
+						"and the angle brackets (< and >) are not allowed."),
+			},
+			"max_app_requests": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"max_ip_requests": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"max_user_requests": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"period_unit": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  timeMinute,
+				ValidateFunc: validation.StringInSlice([]string{
+					timeSecond, timeMinute, timeHour, timeDay,
+				}, false),
+			},
+			"user_throttles": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MaxItems: 30,
+				Elem:     specialThrottleSchemaResource(),
+			},
+			"app_throttles": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MaxItems: 30,
+				Elem:     specialThrottleSchemaResource(),
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  typeExclusive,
+				ValidateFunc: validation.StringInSlice([]string{
+					typeExclusive, typeShared,
+				}, false),
+			},
+			"create_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func specialThrottleSchemaResource() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"max_api_requests": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"throttling_object_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"throttling_object_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func buildApigThrottlingPolicyParameters(d *schema.ResourceData,
+	config *config.Config) (throttles.ThrottlingPolicyOpts, error) {
+	opt := throttles.ThrottlingPolicyOpts{
+		Name:           d.Get("name").(string),
+		TimeInterval:   d.Get("period").(int),
+		TimeUnit:       d.Get("period_unit").(string),
+		ApiCallLimits:  d.Get("max_api_requests").(int),
+		UserCallLimits: d.Get("max_user_requests").(int),
+		AppCallLimits:  d.Get("max_app_requests").(int),
+		IpCallLimits:   d.Get("max_ip_requests").(int),
+		Description:    d.Get("description").(string),
+	}
+	pType := d.Get("type").(string)
+	if val, ok := policyType[pType]; ok {
```

Comment:
The parameter `type` has added a check function in the schema, it is no need to check again.